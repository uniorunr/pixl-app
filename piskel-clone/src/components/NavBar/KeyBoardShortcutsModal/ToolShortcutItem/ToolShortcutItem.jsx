import React, { Component } from 'react';
import PropTypes from 'prop-types';
import './ToolShortcutItem.scss';

class ShortcutItem extends Component {
  activate = () => {
    const { index, makeActive } = this.props;
    makeActive(index);
  };

  render() {
    const {
      iconClass,
      toolName,
      active,
      tools,
      index,
      keyCode,
      id,
    } = this.props;

    return (
      <div
        role="presentation"
        className="shortcuts-modal__row"
        onClick={this.activate}
      >
        <div
          className={`shortcuts-modal__tool ${
            active ? 'shortcuts-modal__tool_active' : ''
          }`}
          data-tool={id}
          ref={(ref) => {
            if (ref) {
              tools[index] = ref;
            }
          }}
        >
          <i className={iconClass} />
          <span>{toolName}</span>
          <span>-</span>
          <span className="shortcuts-modal__shortcut">{keyCode}</span>
        </div>
      </div>
    );
  }
}

ShortcutItem.propTypes = {
  iconClass: PropTypes.string.isRequired,
  toolName: PropTypes.string.isRequired,
  active: PropTypes.bool.isRequired,
  tools: PropTypes.instanceOf(Array).isRequired,
  index: PropTypes.number.isRequired,
  makeActive: PropTypes.func.isRequired,
  keyCode: PropTypes.string.isRequired,
  id: PropTypes.string.isRequired,
};

export default ShortcutItem;
