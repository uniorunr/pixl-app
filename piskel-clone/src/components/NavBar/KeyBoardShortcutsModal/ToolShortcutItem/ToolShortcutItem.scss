.shortcuts-modal__row {
  display: flex;
  align-items: center;
  justify-content: flex-start;

  .shortcuts-modal__tool {
    cursor: pointer;
    border: 2px dashed transparent;
    padding: 4px 10px;

    span {
      margin: 0 10px 0 10px;
    }

    i {
      font-size: 1.2rem;
    }

    input {
      width: 40px;
      margin: 0 10px;
      border: 2px solid black;
      outline: none;
      padding: 2px 5px;
    }

    input:focus {
      border: 2px dashed black;
    }

    button {
      position: relative;
      top: 0;
      background: white;
      border: 2px solid black;
      font-weight: 700;
      padding: 2px 5px;
      outline: none;
      box-shadow: 2px 2px 0 black;
      cursor: pointer;
    }

    button:active {
      top: 2px;
      box-shadow: 0 0 0 black;
    }
  }

  .shortcuts-modal__tool_active {
    border: 2px dashed black;
    padding: 4px 10px;
  }
}
