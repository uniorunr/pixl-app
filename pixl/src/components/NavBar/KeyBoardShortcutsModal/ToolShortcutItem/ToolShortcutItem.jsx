import React, { Component } from 'react';
import { connect } from 'react-redux';
import PropTypes from 'prop-types';
import './ToolShortcutItem.scss';

class ShortcutItem extends Component {
  activate = () => {
    const { index, makeActive } = this.props;
    makeActive(index, 'tools');
  };

  render() {
    const {
      toolsData, active, tools, index, id,
    } = this.props;

    return (
      <div
        role="presentation"
        className="shortcuts-modal__row"
        onClick={this.activate}
      >
        <div
          className={`shortcuts-modal__tool ${
            active ? 'shortcuts-modal__tool_active' : ''
          }`}
          data-tool={id}
          ref={(ref) => {
            if (ref) {
              tools[index] = ref;
            }
          }}
        >
          <i className={toolsData[id] ? toolsData[id].iconClass : ''} />
          <span>{toolsData[id] ? toolsData[id].name : ''}</span>
          <span>-</span>
          <span className="shortcuts-modal__shortcut">
            {toolsData[id] ? toolsData[id].shortcut : ''}
          </span>
        </div>
      </div>
    );
  }
}

ShortcutItem.propTypes = {
  toolsData: PropTypes.instanceOf(Object).isRequired,
  active: PropTypes.bool.isRequired,
  tools: PropTypes.instanceOf(Array).isRequired,
  index: PropTypes.number.isRequired,
  makeActive: PropTypes.func.isRequired,
  id: PropTypes.string.isRequired,
};

const mapStateToProps = state => ({
  tools: state.shortcuts.toolsShortcutsRefs,
  toolsData: state.tools.toolsData,
});

export default connect(mapStateToProps)(ShortcutItem);
