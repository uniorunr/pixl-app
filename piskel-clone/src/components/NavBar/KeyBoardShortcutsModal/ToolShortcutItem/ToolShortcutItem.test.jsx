import React from 'react';
import renderer from 'react-test-renderer';
import ToolShortcutItem from './ToolShortcutItem';

it('renders correctly', () => {
  const tree = renderer
    .create(
      <ToolShortcutItem
        active
        iconClass=""
        id=""
        index={0}
        keyCode=""
        makeActive={() => {}}
        toolName=""
        tools={[]}
      />,
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
