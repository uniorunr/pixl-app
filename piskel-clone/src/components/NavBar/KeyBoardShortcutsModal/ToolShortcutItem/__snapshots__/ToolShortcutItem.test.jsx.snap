// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="shortcuts-modal__row"
  onClick={[Function]}
  role="presentation"
>
  <div
    className="shortcuts-modal__tool shortcuts-modal__tool_active"
    data-tool=""
  >
    <i
      className=""
    />
    <span>
      
    </span>
    <span>
      -
    </span>
    <span
      className="shortcuts-modal__shortcut"
    >
      
    </span>
  </div>
</div>
`;
